src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -502,19 +500,49 @@ private static BoundExpression CreateAnonymousFunctionConversion(SyntaxNode synt
             // UNDONE: is converted to a delegate that does not match. What to surface then?
 
             var unboundLambda = (UnboundLambda)source;
-            var boundLambda = unboundLambda.Bind((NamedTypeSymbol)destination);
-            diagnostics.AddRange(boundLambda.Diagnostics);
+            var expr = createAnonymousFunctionConversion(syntax, unboundLambda, conversion, isCast, conversionGroup, destination, diagnostics);
 
-            return new BoundConversion(
-                syntax,
-                boundLambda,
-                conversion,
-                @checked: false,
-                explicitCastInCode: isCast,
-                conversionGroup,
-                constantValueOpt: ConstantValue.NotAvailable,
-                type: destination)
-            { WasCompilerGenerated = source.WasCompilerGenerated };
+            if (destination.SpecialType == SpecialType.System_Delegate || destination.IsNonGenericExpressionType())
```

Comment:
>destination.SpecialType == SpecialType.System_Delegate [](start = 16, length = 54)

Same here (`object`, `ICloneable`, etc)

Reply:
Let's address later when addressing implicit conversions to those base and interface types.